src/setup_scripts/core_contract.rs

```diff
@@ -0,0 +1,71 @@
+use std::str::FromStr;
+
+use ethers::abi::Address;
+
+use crate::contract_clients::config::Config;
+use crate::contract_clients::core_contract::{CoreContract, CoreContractDeploy, CoreContractStarknetL1Enum};
+use crate::contract_clients::starknet_sovereign::StarknetSovereignContract;
+use crate::contract_clients::starknet_validity::StarknetValidityContract;
+use crate::contract_clients::utils::get_bridge_init_configs;
+use crate::utils::{save_to_json, JsonValueType};
+use crate::CliArgs;
+
+pub struct CoreContractStarknetL1<'a> {
+    arg_config: &'a CliArgs,
```

Comment:
this should be only the relevant args needed

Reply:
clippy suggested it so I added otherwise it was giving the errors

Reply:
what I mean is that we shouldn't pass the entire struct `CliArgs` because it has a lot of irrelevant stuff. we just need to pass those fields which are needed.

Reply:
this is pending

Reply:
skipped